### PR TITLE
feat(tel-ui): Pod connection status and latency

### PR DESCRIPTION
**Pull request by davidbeechey**
- Pod connection status now uses hearbeat with pod
- Latency graph on hover
- Latency value colour-coded
- New context for storing pod state (connection status and latency info)

### CODE REVIEW

**Comment by davidbeechey** [comment]
Changes:
- Fixed the pod disconnect message only appearing if MQTT broker is disconnected and not the heartbeat
- Added tooltip for connection status including uptime and connection established time

**Comment by davidbeechey** [comment]
Added:
- State machine flow chart now has key
- Pod state groups renamed to passive, active and failure states
- Pod state logic moved to context

File: telemetry/packages/ui/app/controls/controls.ts [hunk 1]
```diff
@@ -1,11 +1,9 @@
 import { MqttPublish } from '@hyped/telemetry-types';
 import { toast } from 'react-hot-toast';
 
-export const calibrate = (podId: string) => {
-  console.log('Calibrating');
-  toast(`[${podId}] Calibrating!`);
-};
-
+/**
+ * TODO: Not implemented
```
Comment (arjunnaha) [inline telemetry/packages/ui/app/controls/controls.ts:5]: To implement

Comment (davidbeechey) [reply telemetry/packages/ui/app/controls/controls.ts]: done :)

File: telemetry/packages/ui/app/components/flow/nodes/active-node.tsx
[the remark this answers is not in the document]
Comment (davidbeechey) [reply telemetry/packages/ui/app/components/flow/nodes/active-node.tsx:1]: done :)

File: telemetry/packages/ui/app/context/pods.tsx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      });
+    };
+  }, [client]);
+
+  return (
+    <PodsContext.Provider value={podsState}>{children}</PodsContext.Provider>
+  );
+};
+
+export const usePods = (podId: string) => {
```
[the remark this answers is not in the document]
Comment (davidbeechey) [reply telemetry/packages/ui/app/context/pods.tsx]: done :)

File: telemetry/packages/ui/app/context/pods.tsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+      }
+
+      if (topic !== getTopic('latency/response', podId)) return;
+
+      // calculate the latency
+      const latency =
+        new Date().getTime() -
+        parseInt(JSON.parse(message.toString())['latency']);
+
+      console.log(`Received latency response from ${podId}: ${latency}ms`);
```
[the remark this answers is not in the document]
Comment (davidbeechey) [reply telemetry/packages/ui/app/context/pods.tsx]: removed 👍

**Review by arjunnaha — APPROVED** [review]
LGTM!

Mr. Arjun Naha Esq.
Esteemed Project Manager of the Telemetry Project
Hyperloop Edinburgh
University of Edinburgh
Influencing the world since 1583

**Pull request merged**